pingora-core/src/listeners/mod.rs

```diff
@@ -118,6 +119,17 @@ impl UninitializedStream {
             Ok(Box::new(self.l4))
         }
     }
+
+    /// Get the peer address of the connection if available
+    pub fn peer_addr(&self) -> Option<String> {
+        let stream = &self.l4;
+        {
+            if let Some(socket_digest) = stream.get_socket_digest() {
```

Comment:
nit: this could be more succinctly expressed like `self.l4.get_socket_digest().and_then(|d| d.peer_addr().map(|addr| addr.to_string()))`

Reply:
I'll revert that commit and will amend that function.